ci.md

```diff
@@ -0,0 +1,201 @@
+`push` checks, while pull requests from the official repositories will have
+both `pull_request` and `push`.
+
+Other checks run on GitHub Actions:
+
+* `DCO`: Checks that all commits are signed correctly for the
+    [Developer Certificate of Origin](https://developercertificate.org/).
+* `PR Collection Labeler`: Adds collection labels (i.e. Blueprint, Citadel...)
+    to the pull request automatically according to the target branch.
+* `Ticket opened`: Adds the pull request to the [Core development board](https://github.com/orgs/ignitionrobotics/projects/3).
+* `codecov/*`: Checks that the test coverage hasn't been reduced.
+
+## Required checks
+
+Some checks are marked as `Required`. This means the pull request can't
+be merged unless that check is green.
+
+Ideally, all checks would be required. But known warnings, test failures and
+flaky tests often prevent us from marking them as required. Until these are
+addressed, some checks will need to be checked by hand when they fail, to see
+if those failures are unexpected. See
+[this issue](https://github.com/ignition-tooling/release-tools/issues/398) for
+what checks are required for each library.
+
+## Interpreting results
+
+On the GitHub UI, checks can be:
+
+* ✅: Passed, all good!
+* 🟡: Pending, results haven't been received yet.
+* ❌: Failed, something is wrong.
+
+Depending on the library and on the build queue, checks can be in a pending
+🟡 state from a couple of minutes up to an entire day. If a build is in this
+state for too long, there may be some issue with infrastructure and the
+build didn't run or didn't report back.
+
+Failing builds ❌ usually require more investigation:
+
+* If the build is `Required`, something is definitely wrong and must be fixed
+  before merging the pull request.
+* If the build is not required, investigation is needed to find out if the
+  failure is pre-existing or if it is being introduced by the pull request.
+
+Clicking on the `Details` button of a failing build takes you to the build so
+that you can inspect it.
+
+Builds can fail for a variety of reasons, for example:
+
+* The code failed to compile. This should never be accepted, even in non-required
+  checks, and must be fixed.
+    * Jenkins: When there's a compilation failure, the build is marked red 🔴.
+               On the build's page, click on `Console Output` then scroll until
+               you find the compilation errors under the `compiling` section.
+    * Actions: When there's a compilation failure, the build is marked red ❌.
+               On the build logs, the compilation failure should be under the
+               `make` collapsible.
+* There are test failures. Test failures that aren't pre-existing must be fixed.
+    * Jenkins: When there are test failures, the build is marked yellow 🟡.
+               On the build's page, the failing tests are listed under
+               `Test Result`. You can also check the `Console Output` for the
+               full test logs.
```

Comment:
Should we add a note about how to get to the History page of the test? I found that tip to be very helpful in telling me exactly where to look to determine whether it's a new failure or preexisting. If I remember correctly, there were two or more places that say "History" you can click on, and one of them would give me the graph I want. The page seemed different for different test sometimes, and I don't know why that should be.